network/libp2p_impl.go

```diff
@@ -289,8 +316,52 @@ func (dtnet *libp2pDataTransferNetwork) msgToStream(ctx context.Context, s netwo
 		return err
 	}
 
-	if err := s.SetWriteDeadline(time.Time{}); err != nil {
-		log.Warnf("error resetting deadline: %s", err)
-	}
 	return nil
 }
+
+func (impl *libp2pDataTransferNetwork) Protocol(ctx context.Context, id peer.ID) (protocol.ID, error) {
+	// Check the cache for the peer's protocol version
```

Comment:
@dirkmc You don't need this cache as libp2p maintains it's own cache. You can use `impl.host.Peerstore().FirstSupportedProtocol()` for the cache lookup and then go do the open stream thing it it doesen't have anything for you.